LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/RadioMetadata.cs

```diff
@@ -3,35 +3,7 @@
 
 namespace LoRaWan.NetworkServer.BasicsStation
 {
-    public class RadioMetadata
-    {
-        public DataRateIndex DataRate { get; }
-        public Hertz Frequency { get; }
-        public RadioMetadataUpInfo UpInfo { get; }
+    public record RadioMetadata(DataRateIndex DataRate, Hertz Frequency, RadioMetadataUpInfo UpInfo);
 
-        public RadioMetadata(DataRateIndex dataRate, Hertz frequency, RadioMetadataUpInfo upInfo)
-        {
-            DataRate = dataRate;
-            Frequency = frequency;
-            UpInfo = upInfo;
-        }
-    }
-
-    public class RadioMetadataUpInfo
-    {
-        public RadioMetadataUpInfo(uint antennaPreference, ulong xtime, uint gpsTime, double receivedSignalStrengthIndication, float signalNoiseRatio)
-        {
-            AntennaPreference = antennaPreference;
-            Xtime = xtime;
-            GpsTime = gpsTime;
-            ReceivedSignalStrengthIndication = receivedSignalStrengthIndication;
-            SignalNoiseRatio = signalNoiseRatio;
-        }
-
-        public uint AntennaPreference { get; }
-        public ulong Xtime { get; }
-        public uint GpsTime { get; }
-        public double ReceivedSignalStrengthIndication { get; }
-        public float SignalNoiseRatio { get; }
-    }
+    public record RadioMetadataUpInfo(uint AntennaPreference, ulong Xtime, uint GpsTime, double ReceivedSignalStrengthIndication, float SignalNoiseRatio);
```

Comment:
I would recommend making this a regular record as opposed to a positional record (the latter is good when there's a natural order to fields, but I don't think we can assume that here):

```suggestion
    public sealed record RadioMetadataUpInfo
    {
        public RadioMetadataUpInfo(uint antennaPreference, ulong xtime, uint gpsTime,
                                   double receivedSignalStrengthIndication, float signalNoiseRatio)
        {
            AntennaPreference = antennaPreference;
            Xtime = xtime;
            GpsTime = gpsTime;
            ReceivedSignalStrengthIndication = receivedSignalStrengthIndication;
            SignalNoiseRatio = signalNoiseRatio;
        }

        public uint AntennaPreference { get; init; }
        public ulong Xtime { get; init; }
        public uint GpsTime { get; init; }
        public double ReceivedSignalStrengthIndication { get; init; }
        public float SignalNoiseRatio { get; init; }
    }
```